### DISCUSSION TITLE
Issues importing splink on dataproc serverless cluster

### DISCUSSION THREAD

**Discussion by WillHoylesONS** [moj-analytical-services/splink#2340] [General]
Another colleague in ONS and I have been struggling to import splink into a dataproc serverless instance. We're both essentially using the basic google recommended container image from a dockerfile and just adding Splink to the list of dependencies to install. 

The problem doesn't seem to occur with Splink version 2.x but does with 3.x and 4, and there isn't a problem using Splink on a single VM inside Google Cloud Platform. 

The error that returns is:
```File "/var/dataproc/tmp/srvls-batch-9adefea6-2e73-4eee-b38e-27f0afda8eec/dataproc_splink_test.py", line 15, in <module> from splink.spark.linker import SparkLinker
File "/opt/miniconda3/lib/python3.12/site-packages/splink/__init__.py", line 5, in <module> from splink.internals.datasets import splink_datasets
File "/opt/miniconda3/lib/python3.12/site-packages/splink/internals/datasets/__init__.py", line 270, in <module> class _SplinkDataSets(metaclass=_SplinkDataSetsMeta, datasets=_datasets):
File "/opt/miniconda3/lib/python3.12/site-packages/splink/internals/datasets/__init__.py", line 152, in __new__ cls.cache_dir.mkdir(exist_ok=True)
File "/opt/miniconda3/lib/python3.12/pathlib.py", line 1311, in mkdir os.mkdir(self, mode)
PermissionError: [Errno 13] Permission denied: '/opt/miniconda3/lib/python3.12/site packages/splink/internals/datasets/__splinkdata_cache__'```

**Comment by ADBond** [comment]
Thanks for the report! `.../splink/internals/datasets/__splinkdata_cache__` is the folder that Splink uses to store dummy datasets that are available within Splink (but are not automatically downloaded) - these were introduced in version 3.something. It looks like this error is occurring because the process that is installing Splink doesn't have the necessary permissions for creating a folder in that location.

I'll see if I can tweak things so that you shouldn't get this error in future versions. In the meantime a couple of things that might help you could change in your dockerfile are either creating the folder `'/opt/miniconda3/lib/python3.12/site packages/splink/internals/datasets/__splinkdata_cache__'` manually beforehand if possible, or perhaps running the install process with elevated permissions.

**Reply by WillHoylesONS** [reply]
Thanks! I'll give that a go.
